device-code-flow/DeviceCodeHelper.cs

```diff
@@ -0,0 +1,26 @@
+using System;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using Microsoft.Identity.Client;
+
+public static class DeviceCodeHelper
+{
+    public static Func<DeviceCodeResult,Task> GetDeviceCodeResultCallback()
+    {
+        return deviceCodeResult =>
+        {
+            // This will print the message on the console which tells the user where to go sign-in using 
+            // a separate browser and the code to enter once they sign in.
+            // The AcquireTokenWithDeviceCode() method will poll the server after firing this
+            // device code callback to look for the successful login of the user via that browser.
+            // This background polling (whose interval and timeout data is also provided as fields in the 
+            // deviceCodeCallback class) will occur until:
+            // * The user has successfully logged in via browser and entered the proper code
+            // * The timeout specified by the server for the lifetime of this code (typically ~15 minutes) has been reached
+            // * The developing application calls the Cancel() method on a CancellationToken sent into the method.
+            //   If this occurs, an OperationCanceledException will be thrown (see catch below for more details).
+            Console.WriteLine(deviceCodeResult.Message);
```

Comment:
it seems this class has a newer version but more importantly it is already part of [the module itself](https://www.powershellgallery.com/packages/MSAL.PS/4.37.0.0/Content/internal%5CDeviceCodeHelper.cs). I was wondering whether it is really required.

Reply:
So the link you provided is for MSAL.PS which is a project not supported by Microsoft.
While I was able to get a tutorial working with MSAL.PS I was directed to not use it.

For this tutorial we only want to use Microsoft Authentication Library for .NET (MSAL.NET)

Reply:
Oh I see, if that is not supported by MS sounds like a good idea not to use it for now. 

Please let me mark this one a resolved for now.